test/core/structures/replication-structures.test.js

```diff
@@ -0,0 +1,42 @@
+const test = require('ava')
+const { ReplicationConfiguration } = require('../../../lib/core/structures/replication-structures')
+
+test('replication configuration structure works as expected for non config servers', (t) => {
+  const replicaSetName = 'replica-set-0'
+  const hostnames = ['mongod-0', 'mongod-1', 'mongod-2']
+  const port = 27017
```

Comment:
Does this actually make sense to split the port and hostname? this  would enforce all host's use the same port which isn't thecase if you look  at the generated config.

Can I  suggest 2 things?

1. Rename hostname to host
2. Allow just using the hostname i.e `mongod-0` but this class auto adds the port if its not there to the default port?

Reply:
I guess what I would like to see it being able to do this 

`const hosts = ['mongod-0', 'mongod-2:1234', 'localhost:12345`]`

This allows for pretty flexible dev if we need to but still applying default ports based on configuration.

Reply:
The ports are tied to the respective type of the mongod instance:

- 27017 a normal mongod instance or mongos
- 27018 a shard
- 27019 a config server

It is technically possible to overwrite these, but that would create an expectation, which i would like to avoid. 

This is about the testing, right?  If we cannot make it work otherwise, i can totally change it, but dealing with hostnames + different ports will make everything a little more complicated

Reply:
Yes, it was 100% for testing. and in production I 100% agree with you but since this tool is aimed at dev experience I don't think we should discount this option in the future.

But for now, let's leave it simple.

Reply:
As discussed here this is a (semi) requirement for testing more easily https://github.com/unumotors/mongo-hydra/issues/10#issuecomment-650531081